### PR TITLE
neuralprophet 0.4.2

### PR DESCRIPTION
**Pull request by SC426**
[Jira](https://anaconda.atlassian.net/browse/PKG-914)
[Upstream](https://github.com/ourownstory/neural_prophet/tree/0.4.2)
[Changelog](https://github.com/ourownstory/neural_prophet/releases)
[License](https://github.com/ourownstory/neural_prophet/blob/0.4.2/LICENSE)

**Actions:**

- Update version and sha256
- Update dependencies
- Remove `fix_scripts.patch` as it is no longer needed
- Holidays minimum requirement changed due to the following error:
`FAILED tests/test_regularization.py::test_regularization_holidays - ValueErro...
================== 1 failed, 100 passed, 6 warnings in 22.27s ==================
Tests failed for neuralprophet-0.4.2-py38hca03da5_1.tar.bz2`
- Recipe clean up
- Linter fixes

**Notes:**
- There is an intermittent issue on prefect for windows builds (an error with `tkinter`). Refer to this [push](https://github.com/AnacondaRecipes/neuralprophet-feedstock/pull/3/commits/0d1e5eeac1b7126705f7933db6639811c5ed5aa4) (also this [comment](https://github.com/AnacondaRecipes/neuralprophet-feedstock/pull/3#discussion_r1093525334)) for current build issue.
- There are unsatisfiable dependencies for `s390x`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,67 +1,62 @@
...
     - pytest
-    - pytest-cov
   commands:
-    # neuralprophet 0.3.2 requires dataclasses, which is not installed.
-    # but dataclasses 0.8 is available.
-    #- pip check
-    - pytest -v --cov=./ --cov-report=xml  # [not (py==37 or py==310 or (linux and ppc64le))]
-
+    - pip check
+    - pytest -v tests  # [not (py==37 or py==310 or (linux and ppc64le))]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Do tests still fail for py37, py310 and linux-aarch64?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,67 +1,62 @@
...
-    - fix_scripts.patch
+  url: https://github.com/ourownstory/neural_prophet/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 9dea912dc0ef238ccef30b59fc448a430e0f47818ce20502e662e94e4ddc5dc1
 
 requirements:
   host:
-    - m2-patch  # [win] 
-    - patch     # [not win]
+    - m2-patch  # [win64] 
+    - patch     # [not win64]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: As we do not apply patches anymore we can remove `m2-patch` and `patch`

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml]: I removed for testing. Looks like [Windows](https://cloud.prefect.io/anaconda/flow-run/b4e95830-17b0-4a55-a26f-7d09fdfd400f?logs=&logId=a01177b4-0f4d-4f98-985c-370ef0165a95) is failing.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Only one test failed:
` FAILED tests\test_integration.py::test_dict_input - _tkinter.TclError: invali.`

Comment (danpetry) [reply recipe/meta.yaml]: this is a platform related issue, seems to happen quite frequently. Might be worth reporting it as an issue in rocket-platform and giving Sirius team a heads-up on Slack.

Comment (danpetry) [reply recipe/meta.yaml]: something's up with tkinter (which is a built-in python package)

**Review by danpetry — APPROVED** [review]
All good, thanks!

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,67 +1,62 @@
...
+    - ipywidgets >=7.5.1
+    - plotly >=4.14.3
+    - dataclasses >=0.6  # [py<37]
 
 test:
   imports:
     - neuralprophet
   source_files:
-    - tests/
+    - tests/
```
Comment (marcoesters) [inline recipe/meta.yaml]:
```suggestion
    - tests/
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,67 +1,62 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 build:
-  number: 1
-  # lunarcalendar dependencies currently aren't availabl on s390x
-  skip: true  # [py<36 or win32 or s390x]
-  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  # lunarcalendar dependencies currently aren't available on s390x
```
Comment (marcoesters) [inline recipe/meta.yaml]: lunarcalendar is available on s390x

Comment (SC426) [reply recipe/meta.yaml]: Tried removing the skip and got the following prefect error https://cloud.prefect.io/anaconda/flow-run/74f88eb6-e5a4-45ce-9584-fc3ec603dd31?logs=&logId=11a2b1ff-7859-4668-a9f2-d7a4af696679

`conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-s390x: {"typing_extensions[version='>=3.6.4']", "numpy[version='>=1.16']", 'tqdm', "python[version='>=2.7|>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.6']", "python[version='>=3.10,<3.11.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.11,<3.12.0a0']", 'setuptools', "python[version='>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0']", 'python-dateutil', 'packaging', "python[version='>=3.11,<3.12.0a0|>=3.6']", "python[version='>=2.7|>=3.10,<3.11.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0']", "numpy[version='>=1.16.6,<2.0a0|>=1.19.2,<2.0a0|>=1.21.5,<2.0a0|>=1.21.2,<2.0a0']", "importlib_metadata[version='>=3.6']", "_libgcc_mutex[version='*|0.1',build=main]", "pyparsing[version='>=2.0.2|>=2.0.2,!=3.0.5']", "python-dateutil[version='>=2.1|>=2.7']", 'pytz', "certifi[version='>=2016.9.26']", "tornado[version='>=4.2|>=4.2,<7.0|>=6.1|>=6.1,<7.0']", "python-dateutil[version='>=2.7.3|>=2.8.1']", "python[version='>=3.10,<3.11.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0']", "six[version='>=1.5']", "python[version='>=3.7,<3.8.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0']", "pytz[version='>=2014.10']", "setuptools[version='>=18.5']", "convertdate[version='>=2.3.0']", "pyparsing[version='>=2.0.2|>=2.0.2,!=3.0.5|>=2.0.3,!=2.0.4,!=2.1.2,!=2.1.6|>=2.2.1']", "importlib_metadata[version='>=0.12']", "python[version='>=3.10,<3.11.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.6']", "typing_extensions[version='3.10.0.0|3.10.0.2|4.1.1|4.3.0|4.3.0|4.3.0|4.3.0|4.4.0|3.7.4.3|3.7.4.3|3.7.4.2',build='pyh06a4308_0|pyh06a4308_0|pyh06a4308_0|py38ha847dfd_0|py310ha847dfd_0|py310ha847dfd_0|py38ha847dfd_0|py39ha847dfd_0|py37ha847dfd_0|py37ha847dfd_0|py39ha847dfd_0|pyh06a4308_0|py_0|py_0']", 'tornado', "python[version='>=2.7|>=3.5|>=3|>=3.6|>=3.4']", "packaging[version='>=20.0']", 'typing_extensions', "numpy[version='>=1.16.6,<2.0a0|>=1.19.2,<2.0a0|>=1.21.5,<2.0a0|>=1.21.2,<2.0a0|>=1.21,<2|>=1.19,<2|>=1.19.2,<2|>=1.21.2,<2']", "python[version='>=3.5']", "six[version='1.15.0|>=1.15.0']", "libuuid[version='>=1.0.3,<2.0a0|>=1.41.5,<2.0a0']", "pytz[version='>=2017.2|>=2017.3|>=2020.1']", "pytorch[version='>=0.4.1']", 'python', "_openmp_mutex[version='>=4.5']", "certifi[version='>=2020.06.20']", "python[version='>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.6|>=3.5|>=3.7']", "numpy[version='>=1.16,<2.0a0|>=1.21,<2.0a0|>=1.19.2,<2.0a0|>=1.21.5,<2.0a0|>=1.21.2,<2.0a0|>=1.16.6,<2.0a0']", 'matplotlib-base', "numpy[version='>=1.16.6,<2.0a0|>=1.19,<2|>=1.21,<2|>=1.19.2,<2|>=1.21.2,<2|>=1.21.2,<2.0a0']", 'six', "python[version='>=3.6']", "python[version='>=3.10,<3.11.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0']", 'numpy', "python-dateutil[version='>=2.8.0']", "pyparsing[version='>=2.0.3,!=2.0.4,!=2.1.2,!=2.1.6|>=2.2.1']", "six[version='>=1.7.0']", '_openmp_mutex', "python[version='>=3.11,<3.12.0a0']"}
`

Comment (marcoesters) [reply recipe/meta.yaml]: Do you need to build it for s390x? If so, the next step is to find the problematic dependency

Comment (SC426) [reply recipe/meta.yaml]: The client does not need a build `s390x`. We will close this issue for now and can reopen if the need arises.

Comment (marcoesters) [reply recipe/meta.yaml]: The comment is still not accurate though. Maybe write something along the lines of that there are unsatisfiable dependencies on s390x to warn future developers.

Comment (SC426) [reply recipe/meta.yaml]: Forgot to make a comment in the recipe. It's been updated now.

**Review by marcoesters — APPROVED** [review]

**Pull request merged**